node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -567,25 +651,101 @@ func ConvertScsiIdToNguid(scsiId string) string {
 	return finalNguid
 }
 
-func (o GetDmsPathHelperGeneric) WaitForDmToExist(volumeUuid string, volumeNguid string, maxRetries int, intervalSeconds int) (string, error) {
-	formatTemplate := strings.Join([]string{"%d", "%w"}, mpathdSeparator)
+func (o GetDmsPathHelperGeneric) GetMpathdOutput(identifiers []string,
+	multipathdCommandFormatArgs []string) (string, error) {
+	mpathdOutput, err := o.WaitForDmToExist(identifiers, WaitForMpathRetries,
+		WaitForMpathWaitIntervalSec, multipathdCommandFormatArgs)
+
+	if err != nil {
+		return "", err
+	}
+
+	if mpathdOutput == "" {
+		return "", &MultipathDeviceNotFoundForVolumeError{identifiers[0]}
+	}
+	return mpathdOutput, nil
+}
+
+func (o GetDmsPathHelperGeneric) WaitForDmToExist(identifiers []string, maxRetries int, intervalSeconds int,
+	multipathdCommandFormatArgs []string) (string, error) {
+	formatTemplate := strings.Join(multipathdCommandFormatArgs, mpathdSeparator)
 	args := []string{"show", "maps", "raw", "format", "\"", formatTemplate, "\""}
-	var err error
 	logger.Debugf("Waiting for dm to exist")
 	for i := 0; i < maxRetries; i++ {
-		err = nil
 		out, err := o.executer.ExecuteWithTimeout(TimeOutMultipathdCmd, multipathdCmd, args)
 		if err != nil {
 			return "", err
 		}
 		dms := string(out)
-		if !strings.Contains(dms, volumeUuid) && !strings.Contains(dms, volumeNguid) {
-			err = os.ErrNotExist
-		} else {
-			return dms, nil
+		for _, identifier := range identifiers {
+			if strings.Contains(dms, identifier) {
+				return dms, nil
+			}
 		}
 
 		time.Sleep(time.Second * time.Duration(intervalSeconds))
 	}
-	return "", err
+	return "", os.ErrNotExist
+}
+
+func (o GetDmsPathHelperGeneric) ParseFieldValuesOfIdentifiers(identifiers []string, mpathdOutput string) map[string]bool {
+	dmFieldValues := make(map[string]bool)
+
+	scanner := bufio.NewScanner(strings.NewReader(mpathdOutput))
+	for scanner.Scan() {
+		identifier, dmFieldValue := o.getLineParts(scanner)
+		if o.IsThisMatchedDmFieldValue(identifiers, identifier) {
```

Comment:
why not use `isSameId`?

Reply:
`isSameId` checks exactly the strings,  `IsThisMatchedDmFieldValue` checks if it's contains it

Reply:
why do we need to check if it contains? would adding a "strip" in `isSameId` allow us to use `isSameId` instead?

Reply:
what do you mean by strip?

Reply:
https://pkg.go.dev/strings#TrimSpace

Reply:
I don't quite understand how it can help

Reply:
I take that as a "no" for my second question [here](https://github.com/IBM/ibm-block-csi-driver/pull/491#discussion_r882430317).
still waiting for an answer to my first question

Reply:
I wanted that it will be like [here](https://github.com/IBM/ibm-block-csi-driver/blob/baac709c5ee26c69246a6818e814fcdd08200cc3/node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go#L528)

Reply:
then why didn't you want `isSameId` to be like there?
I'm not sure it's necessary anyway

Reply:
got it